core/textile/sync/task-executors.go

```diff
@@ -172,3 +182,48 @@ func (s *synchronizer) processBucketBackupOff(ctx context.Context, task *Task) e
 
 	return s.deleteAllFilesInPath(ctx, bucket, "")
 }
+
+func (s *synchronizer) processBucketRestoreTask(ctx context.Context, task *Task) error {
+	if err := checkTaskType(task, bucketRestoreTask); err != nil {
+		return err
+	}
+
+	bucket := task.Args[0]
+
+	if err := s.restoreBucket(ctx, bucket); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *synchronizer) processRestoreFile(ctx context.Context, task *Task) error {
```

Comment:
If we create a private function for the iterator at `restore.go` we could maybe call that function from here instead of repeating